### ISSUE TITLE
Error in `java': munmap_chunk(): invalid pointer: 0x00007f014415b1d0

### ISSUE THREAD

**Issue by etcinitd** [mesosphere/marathon#1671]
I am running Mesos 0.22.1 and Marathon 0.9.1-RC1 as docker containers on my MacOS X 10.10.3 VirtualBox 4.3.26 r98988 on a CoreOS version 681.0.0 VM and Mesos is up and working but I am consistently getting the following error on Marathon:  
```
Jun 17 01:45:47 coremaster-1 systemd[1]: Starting Marathon...
Jun 17 01:45:47 coremaster-1 docker[11517]: marathon
Jun 17 01:45:47 coremaster-1 docker[11522]: marathon
Jun 17 01:45:47 coremaster-1 systemd[1]: Started Marathon.
Jun 17 01:45:47 coremaster-1 docker[11533]: MESOS_NATIVE_JAVA_LIBRARY is not set. Searching in /usr/lib /usr/local/lib.
Jun 17 01:45:47 coremaster-1 docker[11533]: MESOS_NATIVE_LIBRARY, MESOS_NATIVE_JAVA_LIBRARY set to '/usr/lib/libmesos.so'
Jun 17 01:45:47 coremaster-1 docker[11533]: [2015-06-17 01:45:47,625] INFO Starting Marathon 0.9.0-RC1 (mesosphere.marathon.Main$:87)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,074] INFO Connecting to Zookeeper... (mesosphere.marathon.Main$:37)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,089] INFO Client environment:zookeeper.version=3.4.6-1569965, built on 02/20/2014 09:09 GMT (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,090] INFO Client environment:host.name=<NA> (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,090] INFO Client environment:java.version=1.7.0_79 (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,091] INFO Client environment:java.vendor=Oracle Corporation (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,092] INFO Client environment:java.home=/usr/lib/jvm/java-7-openjdk-amd64/jre (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,092] INFO Client environment:java.class.path=/marathon/bin/../target/marathon-assembly-0.9.0-RC1.jar (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,093] INFO Client environment:java.library.path=/usr/java/packages/lib/amd64:/usr/lib/x86_64-linux-gnu/jni:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/usr/lib/jni:/lib:/usr/lib (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,093] INFO Client environment:java.io.tmpdir=/tmp (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,094] INFO Client environment:java.compiler=<NA> (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,095] INFO Client environment:os.name=Linux (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,095] INFO Client environment:os.arch=amd64 (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,096] INFO Client environment:os.version=4.0.3 (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,096] INFO Client environment:user.name=root (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,097] INFO Client environment:user.home=/root (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,099] INFO Client environment:user.dir=/marathon (org.apache.zookeeper.ZooKeeper:100)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,100] INFO Initiating client connection, connectString=172.17.8.100:2181 sessionTimeout=10000 watcher=com.twitter.common.zookeeper.ZooKeeperClient$3@55cfffa2 (org.apache.zookeeper.ZooKeeper:438)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,119] INFO Opening socket connection to server 172.17.8.100/172.17.8.100:2181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn:975)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,127] INFO Socket connection established to 172.17.8.100/172.17.8.100:2181, initiating session (org.apache.zookeeper.ClientCnxn:852)
Jun 17 01:45:49 coremaster-1 docker[11533]: [2015-06-17 01:45:49,135] INFO Session establishment complete on server 172.17.8.100/172.17.8.100:2181, sessionid = 0x14dff208b5a00bf, negotiated timeout = 10000 (org.apache.zookeeper.ClientCnxn:1235)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,191] INFO Initiating client connection, connectString=172.17.8.100:2181 sessionTimeout=1800000 watcher=com.twitter.zk.EventBroker@1592bfeb (org.apache.zookeeper.ZooKeeper:438)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,193] INFO Opening socket connection to server 172.17.8.100/172.17.8.100:2181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn:975)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,194] INFO Socket connection established to 172.17.8.100/172.17.8.100:2181, initiating session (org.apache.zookeeper.ClientCnxn:852)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,197] INFO Session establishment complete on server 172.17.8.100/172.17.8.100:2181, sessionid = 0x14dff208b5a00c0, negotiated timeout = 40000 (org.apache.zookeeper.ClientCnxn:1235)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,372] INFO Registering in Zookeeper with hostPort:172.17.8.100:8080 (mesosphere.marathon.MarathonModule:206)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,699] INFO Now standing by. Closing existing handles and rejecting new. (mesosphere.marathon.event.http.HttpEventStreamActor:100)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,761] INFO Adding HTTP support. (mesosphere.marathon.MarathonApp$$anon$1:47)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,762] INFO No HTTPS support configured. (mesosphere.marathon.MarathonApp$$anon$1:50)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,776] INFO jetty-8.y.z-SNAPSHOT (org.eclipse.jetty.server.Server:272)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,836] INFO Starting up (mesosphere.marathon.MarathonSchedulerService:150)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,837] INFO Beginning run (mesosphere.marathon.MarathonSchedulerService:155)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,838] INFO Will offer leadership after 500 milliseconds backoff (mesosphere.marathon.MarathonSchedulerService:326)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,942] INFO Registering com.codahale.metrics.jersey.InstrumentedResourceMethodDispatchAdapter as a provider class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:113)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,960] INFO Registering com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider as a provider class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:113)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,962] INFO Registering mesosphere.marathon.api.MarathonExceptionMapper as a provider class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:113)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,962] INFO Registering mesosphere.marathon.api.v2.AppsResource as a root resource class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:116)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,962] INFO Registering mesosphere.marathon.api.v2.TasksResource as a root resource class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:116)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,962] INFO Registering mesosphere.marathon.api.v2.EventSubscriptionsResource as a root resource class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:116)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,963] INFO Registering mesosphere.marathon.api.v2.QueueResource as a root resource class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:116)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,964] INFO Registering mesosphere.marathon.api.v2.GroupsResource as a root resource class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:116)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,965] INFO Registering mesosphere.marathon.api.v2.InfoResource as a root resource class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:116)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,965] INFO Registering mesosphere.marathon.api.v2.LeaderResource as a root resource class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:116)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,966] INFO Registering mesosphere.marathon.api.v2.DeploymentsResource as a root resource class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:116)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,967] INFO Registering mesosphere.marathon.api.v2.ArtifactsResource as a root resource class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:116)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,967] INFO Registering mesosphere.marathon.api.v2.SchemaResource as a root resource class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:116)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,968] INFO Registering mesosphere.chaos.validation.ConstraintViolationExceptionMapper as a provider class (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:113)
Jun 17 01:45:50 coremaster-1 docker[11533]: [2015-06-17 01:45:50,976] INFO Initiating Jersey application, version 'Jersey: 1.18.1 02/19/2014 03:28 AM' (com.sun.jersey.server.impl.application.WebApplicationImpl:815)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,043] INFO Binding com.codahale.metrics.jersey.InstrumentedResourceMethodDispatchAdapter to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,072] INFO Binding mesosphere.marathon.api.MarathonExceptionMapper to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,074] INFO Binding mesosphere.chaos.validation.ConstraintViolationExceptionMapper to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,076] INFO Binding com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,110] INFO HV000001: Hibernate Validator 5.1.2.Final (org.hibernate.validator.internal.util.Version:27)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,357] INFO Using HA and therefore offering leadership (mesosphere.marathon.MarathonSchedulerService:333)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,369] INFO Set group member ID to member_0000000093 (com.twitter.common.zookeeper.Group:426)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,382] INFO Candidate /marathon/leader/member_0000000093 waiting for the next leader election, current voting: [member_0000000093, member_0000000092] (com.twitter.common.zookeeper.CandidateImpl:165)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,857] INFO Binding mesosphere.marathon.api.v2.AppsResource to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,897] INFO Binding mesosphere.marathon.api.v2.TasksResource to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,913] INFO Binding mesosphere.marathon.api.v2.EventSubscriptionsResource to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,915] INFO Binding mesosphere.marathon.api.v2.QueueResource to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,920] INFO Binding mesosphere.marathon.api.v2.GroupsResource to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,921] INFO Binding mesosphere.marathon.api.v2.InfoResource to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,923] INFO Binding mesosphere.marathon.api.v2.LeaderResource to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,925] INFO Binding mesosphere.marathon.api.v2.DeploymentsResource to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,931] INFO Binding mesosphere.marathon.api.v2.ArtifactsResource to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:51 coremaster-1 docker[11533]: [2015-06-17 01:45:51,933] INFO Binding mesosphere.marathon.api.v2.SchemaResource to GuiceManagedComponentProvider with the scope "Singleton" (com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory:168)
Jun 17 01:45:52 coremaster-1 docker[11533]: [2015-06-17 01:45:52,000] INFO Started SelectChannelConnector@0.0.0.0:8080 (org.eclipse.jetty.server.AbstractConnector:338)
Jun 17 01:45:54 coremaster-1 docker[11533]: [2015-06-17 01:45:54,006] INFO Candidate /marathon/leader/member_0000000093 is now leader of group: [member_0000000093] (com.twitter.common.zookeeper.CandidateImpl:152)
Jun 17 01:45:54 coremaster-1 docker[11533]: [2015-06-17 01:45:54,008] INFO Elected (Leader Interface) (mesosphere.marathon.MarathonSchedulerService:244)
Jun 17 01:45:54 coremaster-1 docker[11533]: *** Error in `java': munmap_chunk(): invalid pointer: 0x00007f014415b1d0 ***
Jun 17 01:45:54 coremaster-1 docker[11533]: /marathon/bin/start.sh: line 12:     8 Aborted                 (core dumped) LIBPROCESS_IP=${NCP_IP} LIBPROCESS_PORT=9090 /marathon/bin/start --hostname ${NCP_IP} --master "zk://${ZK_URL}/mesos" --zk "zk://${ZK_URL}/marathon" --checkpoint --task_launch_timeout 300000 --local_port_min 31000 --local_port_max 32000
Jun 17 01:45:54 coremaster-1 systemd[1]: marathon.service: main process exited, code=exited, status=134/n/a
Jun 17 01:45:54 coremaster-1 docker[11612]: marathon
Jun 17 01:45:54 coremaster-1 systemd[1]: Unit marathon.service entered failed state.
Jun 17 01:45:54 coremaster-1 systemd[1]: marathon.service failed.
```

**Comment by aquamatthias** [comment]
Hey @etcinitd The java virtual machine is dying here. Since you running this from OSX: how many resources are assigned to the linux docker machine? Can you assign more memory and try again?

**Comment by etcinitd** [comment]
Hi Mattias,
I tested it with both 1GB and 2GB of ram for the CoreOS vm on Virtual Box
and both failed with the same error. Then, I enabled debug logs on marathon
which shows it's failing when trying to load a native mesos library. I then
tested it on a Windows 8 laptop with the same version of virtual box and it
works! I'm now curious to see what that native mesos library is trying to
do that seems to have some issues with Mac.

**Comment by aquamatthias** [comment]
Hi @etcinitd  thanks for trying that out. 
I tried docker-machine on OSX (this will run the https://github.com/boot2docker/boot2docker distribution) and started the image with: `docker run -p 8080:8080  mesosphere/marathon:v0.9.0-RC1 --master mylocalmachineip:5050 --zk zk://mylocalmachineip:2181/marathon` 
The image started successfully.

Can you give some directions, how to reproduce the problem?

**Comment by etcinitd** [comment]
Hi Mattias,
Thanks for your followup. I also tested it with boot2docker and it worked! So, I'm now suspicious that something can be incompatible here with CoreOS. 
As our setup is a bit complicated, I have pushed a pruned test version for you here, so that you can reproduce it easily: https://github.com/etcinitd/test-marathon-coreos
Feel free to let me know if you have any questions on how to get it up and running.

**Comment by aquamatthias** [comment]
Hi @etcinitd thanks for providing this setup, which is awesome!
I think this issue is not a problem of Marathon, since we see a crashing JVM, but a CoreOS integration problem. Since I am not an expert in CoreOS, I hope somebody can look into this issue.

**Comment by boynux** [comment]
The issue is you need to add your box name (`coremaster-1` in this case) to Docker host `/etc/hosts` to make it available in Docker container. It's failing because it can't get containers IP address based on host's `hostname` and it crashes. I know this is strange but it's the way Docker works in `host` networking mode.

Should it be solved in Marathon code or it's an expected behavior, I don't know! But I recommend to Marathon guys to provide a fall back mechanism in such conditions (perhaps just to use 127.0.0.1). Or at least document it somewhere.

**Comment by aquamatthias** [comment]
Thanks for the report @boynux. Since the jvm is crashing, I suspect the problem occurs on the mesos JNI Layer. I will speak to some mesos guys. But I agree with your suggestion - if there is no hostname, a sensible default should be taken.

**Comment by aquamatthias** [comment]
This is the related mesos bug: https://issues.apache.org/jira/browse/MESOS-3145 It is fixed in mesos 0.24.0.

In the mean time we can try to fix that in Marathon.
Can you give us the info, what the result is of that statement:
```java.net.InetAddress.getLocalHost.getHostName```
